### PR TITLE
Added Color Picker

### PR DESCRIPTION
**Pull request by andrewsilke**
layouts.labelColorBtn which is a class for a label and a color picker button

### CODE REVIEW

File: zoo/libs/pyqt/widgets/layouts.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from zoo.libs.pyqt.qt import QtWidgets, QtCore, QtGui
 from zoo.libs.pyqt.extended import combobox
-
+from PySide2.QtGui import QColor
```
Comment (dsparrow27) [inline zoo/libs/pyqt/widgets/layouts.py]: no need to import the qcolor class

File: zoo/libs/pyqt/widgets/layouts.py [hunk 2]
```diff
@@ -286,3 +286,57 @@ def __init__(self,  okText="OK", cancelTxt="Cancel", parent=None):
...
+        :param initialRgbColorF: The initial rgb color in 0-1.0 ranges, if None defaults to initialRgbColor values
+        :type initialRgbColorF: tuple
+        :param parent: the widget parent
+        :type parent: class
+        """
+        super(labelColorBtn, self).__init__(parent=parent)
+        self.layout = QtWidgets.QHBoxLayout()
+        self.layout.addWidget(QtWidgets.QLabel(label, parent=self))
+        self.colorPickerBtn = QtWidgets.QPushButton("", parent=self)
+        if not initialRgbColorF:  # if initialRgbColorF is None then input values are in 0-255 range
```
Comment (dsparrow27) [inline zoo/libs/pyqt/widgets/layouts.py]: self.storeRgbColor = initialRgbColor or tuple([i*255 for i in initialRgbColorF])?

File: zoo/libs/pyqt/widgets/layouts.py [hunk 3]
```diff
@@ -286,3 +286,57 @@ def __init__(self,  okText="OK", cancelTxt="Cancel", parent=None):
...
+    def pickColor(self):
+        """Opens the color picker window
+        If ok is pressed then the new color is returned in 0-255 ranges Eg (128, 255, 12)
+        If Cancel is pressed the color is invalid and nothing happens
+        """
+        initialPickColor = QColor(self.storedRgbColor[0], self.storedRgbColor[1], self.storedRgbColor[2], 255)
+        color = QtWidgets.QColorDialog.getColor(initialPickColor)
+        if QColor.isValid(color):  # if the user hits cancel the returned color is invalid, so don't update
+            self.storedRgbColor = (color.getRgb())[0:3]
+            self.colorPickerBtn.setStyleSheet("background-color: {}".format(color.name()))
```
Comment (dsparrow27) [inline zoo/libs/pyqt/widgets/layouts.py]: Nice use of the style sheet

File: zoo/libs/pyqt/widgets/layouts.py [hunk 4]
```diff
@@ -286,3 +286,57 @@ def __init__(self,  okText="OK", cancelTxt="Cancel", parent=None):
...
+
+    def rgbColor(self):
+        """returns rgb tuple with 0-255 ranges Eg (128, 255, 12)
+        """
+        return self.storedRgbColor
+
+    def rgbColorF(self):
+        """returns rgb tuple with 0-1.0 float ranges Eg (1.0, .5, .6666)
+        """
+        return (float(self.storedRgbColor[0])/255, float(self.storedRgbColor[1])/255, float(self.storedRgbColor[2])/255)
```
Comment (dsparrow27) [inline zoo/libs/pyqt/widgets/layouts.py]: tuple([float(i)/255] for i in self.storeRgbColor]) ?

File: zoo/libs/pyqt/widgets/layouts.py (hunk 1 above)
Comment (andrewsilke) [inline zoo/libs/pyqt/widgets/layouts.py]: yep should have sen that one

File: zoo/libs/pyqt/widgets/layouts.py (hunk 2 above)
Comment (andrewsilke) [inline zoo/libs/pyqt/widgets/layouts.py]: yep nice, good way

File: zoo/libs/pyqt/widgets/layouts.py (hunk 3 above)
Comment (andrewsilke) [inline zoo/libs/pyqt/widgets/layouts.py]: ta :)

File: zoo/libs/pyqt/widgets/layouts.py (hunk 4 above)
Comment (andrewsilke) [inline zoo/libs/pyqt/widgets/layouts.py]: nice

**Comment by andrewsilke** [comment]
Done

File: zoo/libs/pyqt/widgets/layouts.py (hunk 2 above)
Comment (dsparrow27) [inline zoo/libs/pyqt/widgets/layouts.py]: i meant this is the code that would replace the if and else

basically 

`initialRgbColor or tuple([i*255 for i in initialRgbColorF])`
 means that if initialRgbColor is False,None 0  then the second statement is used instead. its a good way of shortening code. it should have the same result that you've got but clearer.

Comment (andrewsilke) [inline zoo/libs/pyqt/widgets/layouts.py]: sheesh really? if or else eh?  yeah I can see that now, good stuff.

**Pull request merged**